### PR TITLE
proxy: ensure response is not nil before sanitizing

### PR DESCRIPTION
**Pull request by bhowe34**
### Description
What does this PR do?

In rare cases the vault client can return a nil response and nil error. In this case that would cause the vault proxy to panic because`sanitizeAutoAuthTokenResponse()` assumes the response isn't nil.

### CODE REVIEW

File: command/agentproxyshared/cache/handler.go [hunk 1]
```diff
@@ -85,6 +85,10 @@ func ProxyHandler(ctx context.Context, logger hclog.Logger, proxier Proxier, inm
 				logical.RespondError(w, http.StatusInternalServerError, fmt.Errorf("failed to get the response: %w", err))
 			}
 			return
+		} else if resp == nil {
+			metrics.IncrCounter([]string{"agent", "proxy", "error"}, 1)
+			logical.RespondError(w, http.StatusInternalServerError, fmt.Errorf("failed to get the response: %w", err))
```
Comment (bhowe34) [inline command/agentproxyshared/cache/handler.go:90]: This metric and response is taken from a similar check a few lines above.

**Comment by akshya96** [comment]
Hi @bhowe34! This looks great to me. Do you know if there are any known bugs or issues related to this PR?

**Comment by bhowe34** [comment]
Hi, none that I'm aware of. The error response is the same as the condition a few lines above, so if there haven't been any issues with that I wouldn't expect any with this.

**Review by akshya96 — APPROVED** [review]
LGTM!

**Pull request merged**